services/src/datasets/external/nfdi/mod.rs

```diff
@@ -665,23 +665,56 @@ mod tests {
     use crate::datasets::external::nfdi::{
         ExpiringDownloadLink, NFDIDataProvider, NFDIDataProviderDefinition,
     };
-    use geoengine_datatypes::dataset::DatasetProviderId;
-    use scienceobjectsdb_rust_api::sciobjectsdbapi::models::v1::{Dataset, Metadata};
+    use futures::StreamExt;
+    use geoengine_datatypes::dataset::{DatasetId, DatasetProviderId, ExternalDatasetId};
+    use httptest::responders::status_code;
+    use httptest::{Expectation, Server};
+    use scienceobjectsdb_rust_api::sciobjectsdbapi::models::v1::{
+        Dataset, Metadata, Object, ObjectGroup,
+    };
+    use scienceobjectsdb_rust_api::sciobjectsdbapi::services::v1::{
+        CreateDownloadLinkResponse, GetDatasetObjectGroupsResponse, GetDatasetResponse,
+        GetProjectDatasetsResponse,
+    };
     use serde_json::Value;
     use std::str::FromStr;
+    use tokio::fs::File;
+    use tokio::io::AsyncReadExt;
+
+    use crate::datasets::listing::{DatasetListOptions, ExternalDatasetProvider, OrderBy};
+    use crate::util::user_input::Validated;
+
+    mod wiremock_gen {
+        // hello.Greeter: is the prefix of all rpc,
+        // MyMockServer: name of the generated Server,
+        wiremock_grpc::generate!("api.services.v1", TestProjectServer);
+    }
```

Comment:
It looks strange that the `mod` definition is surrounded by all the imports. I guess it should be after the imports.

Reply:
96a84faeea669c364687b09127ee5ba955cc163f
But the imports of the generated code must occur after the module definition;